t/python/t0010-job.py

```diff
@@ -683,7 +690,21 @@ def cancel_on_start(future, jobid):
                 }
             ),
         )
-
+        self.assertEqual(
+            result[ids[5]].get_info(),
+            JobInfo(
+                {
+                    "id": ids[5],
+                    "result": flux.constants.FLUX_JOB_RESULT_FAILED,
+                    "t_submit": 0.0,
+                    "exception_occurred": True,
+                    "exception_type": "test",
+                    "exception_note": "",
+                    "exception_severity": 0,
+                }
+            ),
+        )
+        self.assertEqual(result[ids[5]].get_info().returncode, 1)
```

Comment:
why split out the "returncode" check into a separate check, could stick in the first check?  Dunno if subtlety I'm missing.

Reply:
This makes the check explicit since returncode is a derived property

Reply:
I pushed a version with a comment before this test to state the reason for the separate check.